pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -366,7 +366,10 @@ func (gc *GarbageCollector) attemptToDeleteWorker(ctx context.Context, item inte
 		// a cluster-scoped object referring to a namespaced owner is an error that will not resolve on retry, no need to requeue this node
 		return forgetItem
 	} else if err != nil {
-		if _, ok := err.(*restMappingError); ok {
+		if errors.IsConflict(err) {
+			// either a uid or resourceVersion precondition failed, which means the live object changed between lookup/verification and delete.
+			klog.V(5).Infof("error syncing item %s: %v", n, err)
```

Comment:
why we would like to retry? it will not keep failing?

Reply:
if the conflict was because of a uid change, we want to retry to hit this path:

https://github.com/kubernetes/kubernetes/blob/00cd2ae3bc902f8c7a116e9602acf2b486a8f38f/pkg/controller/garbagecollector/garbagecollector.go#L525-L529

if the conflict was because of a resourceVersion change, we want to retry to re-fetch the live item